File: sample-projects/java/src/main/java/de/sots/cellarsandcentaurs/adapter/persistence/PersistedCreatures.java
package de.sots.cellarsandcentaurs.adapter.persistence;

import de.sots.cellarsandcentaurs.application.CreatureFacade;
import de.sots.cellarsandcentaurs.domain.model.Creature;
import de.sots.cellarsandcentaurs.domain.model.CreatureId;
import de.sots.cellarsandcentaurs.domain.model.NoSuchCreatureException;
import de.sots.cellarsandcentaurs.domain.service.Creatures;

public class PersistedCreatures implements Creatures {
    private final CreatureRepository repository;

    public PersistedCreatures(CreatureRepository repository) {
        this.repository = repository;
    }

    @Override
    public void save(Creature creature) {
        repository.save(new CreatureEntity(creature.getId().id(), creature.getType().name()));
    }

    @Override
    public Creature find(CreatureId id) throws NoSuchCreatureException {
        var creatureEntity = repository.findById(id.id()).orElseThrow(() -> new NoSuchCreatureException(id));
        return new Creature(new CreatureId(creatureEntity.getId()), CreatureFacade.STANDARD_CREATURE_TYPE);
    }
}

File: sample-projects/javascript/src/de/sots/cellarsandcentaurs/adapter/persistence/PersistedCreatures.js
import { CreatureRepository } from "./CreatureRepository.js";
import { CreatureEntity } from "./CreatureEntity.js";
import { Creatures } from "../../domain/service/Creatures.js";
import { Creature } from "../../domain/model/Creature.js";
import { CreatureId } from "../../domain/model/CreatureId.js";
import { NoSuchCreatureException } from "../../domain/model/NoSuchCreatureException.js";
import { CreatureFacade } from "../../application/index.js";

export class PersistedCreatures extends Creatures {
    /**
     * @param {CreatureRepository} repository
     */
    constructor(repository) {
        super();
        this.repository = repository;
    }

    async save(creature) {
        await this.repository.save(new CreatureEntity(creature.getId().id));
    }

    async find(id) {
        const creatureEntity = await this.repository.findOne(id.id);
        if (!creatureEntity) {
            throw new NoSuchCreatureException(id);
        }
        return new Creature(new CreatureId(creatureEntity.id), CreatureFacade.STANDARD_CREATURE_TYPE);
    }
}
